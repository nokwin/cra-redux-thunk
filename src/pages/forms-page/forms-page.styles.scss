.forms-page {
  display: flex;
  justify-content: center;
}
