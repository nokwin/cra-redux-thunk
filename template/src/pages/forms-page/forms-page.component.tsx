import React from "react";
import { useFormik } from "formik";
import * as Yup from "yup";

import "./forms-page.styles.scss";

export const FormsPage = () => {
  const formik = useFormik({
    initialValues: {
      email: "",
      password: "",
    },
    onSubmit: (values) => {
      alert(JSON.stringify(values));
    },
    validationSchema: Yup.object({
      email: Yup.string()
        .email("Must be valid email!")
        .required("Email is required!"),
      password: Yup.string()
        .min(8, "Password must be at least 8 symbols!")
        .required("Password is required!"),
    }),
  });

  return (
    <div className="forms-page">
      <form onSubmit={formik.handleSubmit} noValidate>
        <label htmlFor="email">Email Address</label>{" "}
        <input
          id="email"
          name="email"
          type="email"
          onChange={formik.handleChange}
          value={formik.values.email}
        />
        {formik.touched.email && formik.errors.email ? (
          <div>{formik.errors.email}</div>
        ) : null}
        <br /> <br />
        <label htmlFor="password">Password</label>{" "}
        <input
          id="password"
          name="password"
          type="password"
          onChange={formik.handleChange}
          value={formik.values.password}
        />
        {formik.touched.password && formik.errors.password ? (
          <div>{formik.errors.password}</div>
        ) : null}
        <br />
        <br />
        <button type="submit">Submit</button>
      </form>
    </div>
  );
};
