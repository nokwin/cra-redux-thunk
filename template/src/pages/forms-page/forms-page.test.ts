import { FormsPage } from "./forms-page.component";

describe("Forms page component", () => {
  it("should render forms page component", () => {
    expect(FormsPage).toMatchSnapshot();
  });
});
