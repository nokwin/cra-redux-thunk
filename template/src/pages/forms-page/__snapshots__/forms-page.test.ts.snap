// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms page component should render forms page component 1`] = `[Function]`;
